Libraries/PeriphDrivers/Source/SPI/spi_ai87_v2.c

```diff
@@ -0,0 +1,920 @@
+    mxc_spi_target_t target;
+    mxc_spi_init_t init;
+    target.index = req->ssIdx;
+
+    init.use_dma = true;
+    init.dma = MXC_DMA;
+
+    // More overhead, but this function will initalize DMA if not initialized.
+    if (MXC_SPI_DMA_GetInitialized(req->spi) == false) {
+        error = MXC_SPI_DMA_Init(&init);
+        if (error != E_NO_ERROR) {
+            return error;
+        }
+    }
+
+    error = MXC_SPI_SetRegisterCallback(req->spi, req->callback, req->callback_data);
+    if (error != E_NO_ERROR) {
+        return error;
+    }
+
+    error = MXC_SPI_DMA_SetRequestSelect(req);
+    if (error != E_NO_ERROR) {
+        return error;
+    }
+
+    return MXC_SPI_RevA2_ControllerTransactionDMA((mxc_spi_reva_regs_t *)(req->spi), req->tx_buffer,
+                                                  req->tx_len, req->rx_buffer, req->rx_len,
+                                                  req->deassert, &target);
+}
+
+int MXC_SPI_ControllerTransaction(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
+                                  uint8_t *rx_buffer, uint32_t rx_fr_len, uint8_t deassert,
+                                  mxc_spi_target_t *target)
+{
+    return MXC_SPI_RevA2_ControllerTransaction((mxc_spi_reva_regs_t *)spi, tx_buffer, tx_fr_len,
+                                               rx_buffer, rx_fr_len, deassert, target);
+}
+
+int MXC_SPI_ControllerTransactionB(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
+                                   uint8_t *rx_buffer, uint32_t rx_fr_len, uint8_t deassert,
+                                   mxc_spi_target_t *target)
+{
+    return MXC_SPI_RevA2_ControllerTransactionB((mxc_spi_reva_regs_t *)spi, tx_buffer, tx_fr_len,
+                                                rx_buffer, rx_fr_len, deassert, target);
+}
+
+int MXC_SPI_ControllerTransactionDMA(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
+                                     uint8_t *rx_buffer, uint32_t rx_fr_len, uint8_t deassert,
+                                     mxc_spi_target_t *target)
+{
+    return MXC_SPI_RevA2_ControllerTransactionDMA((mxc_spi_reva_regs_t *)spi, tx_buffer, tx_fr_len,
+                                                  rx_buffer, rx_fr_len, deassert, target);
+}
+
+int MXC_SPI_ControllerTransactionDMAB(mxc_spi_regs_t *spi, uint8_t *tx_buffer, uint32_t tx_fr_len,
+                                      uint8_t *rx_buffer, uint32_t rx_fr_len, uint8_t deassert,
+                                      mxc_spi_target_t *target)
+{
+    return MXC_SPI_RevA2_ControllerTransactionDMAB((mxc_spi_reva_regs_t *)spi, tx_buffer, tx_fr_len,
+                                                   rx_buffer, rx_fr_len, deassert, target);
+}
+
+int MXC_SPI_SlaveTransaction(mxc_spi_req_t *req)
+{
+    return E_NOT_SUPPORTED;
+}
+
+int MXC_SPI_SlaveTransactionAsync(mxc_spi_req_t *req)
+{
+    return E_NOT_SUPPORTED;
+}
+
+int MXC_SPI_SlaveTransactionDMA(mxc_spi_req_t *req)
```

Comment:
Simply because they haven't been implemented yet in SPI v2. A future update will support target transactions, but I did not want to add any new functions that SPI v2 will not support.